src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -293,9 +293,9 @@ private void CreateNodeCaches()
                 return new CopiedMethodILNode((EcmaMethod)method);
             });
 
-            _copiedFieldRvas = new NodeCache<EcmaField, CopiedFieldRvaNode>(ecmaField =>
+            _copiedFieldRvas = new NodeCache<int, CopiedFieldRvaNode>(rva =>
```

Comment:
I think this should use module+rva as the key to account for the expected composite r2r outputs (where we would compile multiple IL files into one output file).

Reply:
Yes, good point